src/os/bluestore/BlueFS.cc

```diff
@@ -530,19 +530,20 @@ void BlueFS::dump_block_extents(ostream& out)
   }
 }
 
-int BlueFS::get_block_extents(unsigned id, interval_set<uint64_t> *extents)
+void BlueFS::foreach_block_extents(
+  unsigned id,
+  std::function<void(uint64_t, uint32_t)> fn)
 {
   std::lock_guard nl(nodes.lock);
   dout(10) << __func__ << " bdev " << id << dendl;
   ceph_assert(id < alloc.size());
   for (auto& p : nodes.file_map) {
     for (auto& q : p.second->fnode.extents) {
       if (q.bdev == id) {
-        extents->insert(q.offset, q.length);
+        fn(q.offset, q.length);
```

Comment:
I don't understand what do we gain by this change.
The logic is exactly like before trusting that nodes.lock guarantees a consistent view of BlueFS allocation with the shared-allocator.
In my testing it seems that there might be some in-flight BlueFS freed-allocations which are falsely showing as being allocated by the shared-allocator and which are *still* reported as being allocated by get_block_extents